unit/java_bytecode/generate_concrete_generic_type/generate_java_generic_type.cpp

```diff
@@ -0,0 +1,144 @@
+/*******************************************************************\
+
+ Module: Unit tests for generating new type with generic parameters
+         substitued for concrete types
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <catch.hpp>
+
+#include <util/symbol_table.h>
+
+#include <java_bytecode/generate_java_generic_type.h>
+#include <java_bytecode/java_types.h>
+
+#include <iostream>
+#include <java_bytecode/java_bytecode_language.h>
+#include <language.h>
+
+SCENARIO(
+  "generate_java_generic_type",
+  "[CORE][java_bytecode][generate_java_generic_type][!mayfail]")
+{
+  std::unique_ptr<languaget> java_lang(new_java_bytecode_language());
+
+  // Configure the path loading
+  cmdlinet command_line;
+  command_line.set(
+    "java-cp-include-files",
+    "./java_bytecode/generate_concrete_generic_type");
+  config.java.classpath.emplace_back(
+    "./java_bytecode/generate_concrete_generic_type");
+
+  // Configure the language
+  null_message_handlert message_handler;
+  java_lang->get_language_options(command_line);
+  java_lang->set_message_handler(message_handler);
+
+  std::istringstream java_code_stream("ignored");
+
+  GIVEN("A generic java type and a concrete substitution")
+  {
+    symbol_tablet symbol_table;
+
+    generate_java_generic_typet type_generator(message_handler);
+
+    auto boxed_type_reference = symbol_typet("java.lang.Integer");
+    auto boxed_parameter = java_generic_inst_parametert(boxed_type_reference);
+    auto list_type_reference = java_type_from_string("Ljava/util/List;", "List");
```

Comment:
the second parameter makes not much sense here, its idea is to serve as a prefix when a type variable is generated, to make those unique, as for example `T` could be used as type paramter name in different generic classes